### PR TITLE
Updating listings_controller.rb and fixing issue :->  17683

### LINKED ISSUE #17683: NameError: undefined local variable or method `unpublish_listing`
The following code calls methods that don't exist in the controller:

https://github.com/forem/forem/blob/8a4e1ebeba92c29dbf7cf2bf292c5516cd29b687/app/controllers/admin/listings_controller.rb#L51-L54

This is likely related to work done in forem/forem#16221

Referenced in https://app.honeybadger.io/projects/66984/faults83837532

There exists a [app/controllers/concerns/listings_toolkit.rb](https://github.com/forem/forem/blob/8a4e1ebeba92c29dbf7cf2bf292c5516cd29b687/app/controllers/concerns/listings_toolkit.rb) which used to be mixed into the Admin::ListingsController but is no longer.  It is still mixed into the [app/controllers/listings_controller.rb](https://github.com/forem/forem/blob/8a4e1ebeba92c29dbf7cf2bf292c5516cd29b687/app/controllers/listings_controller.rb)

The impact appears limited to an admin; they are not able to unpublish a listing.

### PR DESCRIPTION
**Pull request by MdSahil-oss**
## What type of PR is this? (check all applicable)

 ✅ Bug Fix

## Description
This PR fixes #17683 .

## Related Tickets & Documents

- Related Issue #17683

## QA Instructions, Screenshots, Recordings
This is a code error fix so no need to show any screenshots, Recordings.

### UI accessibility concerns?
No

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
 ✅ I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/admin/listings_controller.rb [hunk 1]
```diff
@@ -49,8 +49,9 @@ def listing_params
     end
 
     def handle_publish_status
-      unpublish_listing if listing_params[:published] == "0"
-      publish_listing if listing_params[:published] == "1"
+      @listing = Listing.find(params[:id])
```
Comment (jeremyf) [inline app/controllers/admin/listings_controller.rb:52]: Thanks for the contribution.  We already have the `@listing` from the `create` method.  We shouldn't need to re-find the object.

Were you able to run this in the application to verify behavior?

**Review by jeremyf — CHANGES_REQUESTED** [review]

File: spec/requests/admin/listings_spec.rb [hunk 2]
```diff
@@ -27,6 +27,30 @@
...
+        expect do
+          put admin_listing_path(id: listing.id), params: {
+            listing: { published: "0" }
+          }
+        end.to change { listing.reload.published? }.from(true).to(false)
+      end
+    end
+
+    context "when attempting to publish" do
+      let!(:listing) { create(:listing, published: true) }
```
Comment (jeremyf) [inline spec/requests/admin/listings_spec.rb:43]:
```suggestion
      let!(:listing) { create(:listing, published: false) }
```

File: spec/requests/admin/listings_spec.rb [hunk 3]
```diff
@@ -27,6 +27,30 @@
...
+
+      it "publishes the listing" do
+        expect do
+          put admin_listing_path(id: listing.id), params: {
+            listing: { published: "0" }
```
Comment (jeremyf) [inline spec/requests/admin/listings_spec.rb:48]:
```suggestion
            listing: { published: "1" }
```

File: spec/requests/admin/listings_spec.rb [hunk 4]
```diff
@@ -27,6 +27,30 @@
...
+          }
+        end.to change { listing.reload.published? }.from(true).to(false)
```
Comment (jeremyf) [inline spec/requests/admin/listings_spec.rb:50]:
```suggestion
        end.to change { listing.reload.published? }.from(false).to(true)
```

**Comment by jeremyf** [comment]
Thank you for the contribution; I have some suggestions as we want to test going from published to unpublished and unpublished to published.

**Comment by jeremyf** [comment]
@MdSahil-oss we had a broken test that was wrecking havoc on the build.  Could you merge in those changes from the HEAD of the main branch?

**Comment by MdSahil-oss** [comment]
> @MdSahil-oss we had a broken test that was wrecking havoc on the build. Could you merge in those changes from the HEAD of the main branch?

Do you mean I need to merge update-listing_controller.rb branch into my main branch to remove this test failure

**Comment by jeremyf** [comment]
> > @MdSahil-oss we had a broken test that was wrecking havoc on the build. Could you merge in those changes from the HEAD of the main branch?
> 
> @jeremyf Do you mean I need to merge update-listing_controller.rb branch into my main branch to remove this test failure

The other way.  You want to merge forem/forem's latest `main` into your `update-listing_controller.rb` branch.  And then you'd push those changes back up to `MdSahil-oss:update-listing_controller.rb`.

**Review by jeremyf — APPROVED** [review]

**Comment by jeremyf** [comment]
@MdSahil-oss thank you for your contributions; the build should go green and we need one more approval.

**Review by maestromac — APPROVED** [review]
LGTM! Thank you for your contribution @MdSahil-oss !

**Pull request merged**